### DISCUSSION TITLE
How to serialize / compare a model as a parent class?

### DISCUSSION THREAD

**Discussion by xelhark** [pydantic/pydantic#9487] [Question] [answered]
I've got this structure:

```python
from pydantic import BaseModel


class Parent(BaseModel):
    foo: str


class Child(Parent):
    bar: str


if __name__ == '__main__':
    parent = Parent(foo="foo")
    child = Child(foo="foo", bar="bar")
    child_as_parent = Parent.model_validate(child) # This is incorrect as it's returning a child instance

    assert parent == child_as_parent
```

So what's happening here is that I have a parent and a child class. I would like to compare them for their "comparable" parts.
But when I run this line:

```python
child_as_parent = Parent.model_validate(child)
```

I still get a `Child` instance.
Now, I understand why this happens, I could fix this by doing a more basic serialization with something like:

```python
child_as_parent = Parent.model_validate(child.model_dump())
```

But it feels wrong to deserialize and re-serialize and validate everything, given that the child instance is already a valid `Parent`.
Is there any way I can get a simple `Parent` object from a given `Child` instance so I can compare it successfully with the `parent` object without having to serialize / validate the whole object again?

**Comment by uriyyo** [comment, marked as answer]
Hi @xelhark,

You can add `model_config` `revalidate_instances` and everything should work as expected:
```py
from pydantic import BaseModel


class Parent(BaseModel):
    foo: str

    model_config = {
        "revalidate_instances": "subclass-instances",
    }


class Child(Parent):
    bar: str


if __name__ == "__main__":
    parent = Parent(foo="foo")
    child = Child(foo="foo", bar="bar")
    child_as_parent = Parent.model_validate(child)

    assert parent == child_as_parent
```
